cip25-wasm/src/lib.rs

```diff
@@ -14,12 +16,10 @@ pub type PolicyIdV1 = String64;
 pub type PolicyIdV2 = Vec<u8>;
```

Comment:
Do we want these to be straight Strings/Byte arrays? Or are these meant to be multiasset policy IDs/asset names that we should try and integrate with the rest of the library? If so then maybe the v1 vs v2 difference could be abstracted away entirely as a serialization detail and we could probably merge the V1 and V2 structs in general in this library. We could also merge `CIP25Metadata` with `LabelMetadata` to make the API simpler. At that point we'd be changing a lot of serialization code though and it'd no longer match up very well with the codegen output.

Reply:
I would keep these matching the CDDL as-is. Anything we need to wrap to create a better API should be done as a separate utility type in a separate file since the goal is to never modify the codegen files

Reply:
For simply wrapping that's fine, but if we want to integrate the policy IDs/asset names with the other types (are they the same right?) we could *massively* simplify the cip25 crate doing that. It would replace ~12 types with only a couple as we could abstract away the v1 vs v2 difference into just encoding differences and a simple enum at the top level that toggles between text vs byte encoding at lower levels. Speaking from our experience in CSL we never once had to do any complete or even significant regeneration after we've finalized the format as any era difference was only just generating the differences and integrating that with the existing code. Or now that we're generating it's because it's such a massive change (split wasm/rust) that any wrappers would be mostly invalidated anyway. I agree that we should reduce the editing of code generated code and use wrappers when it makes sense, but I don't think it can be avoided here unless we like having awkward APIs with many redundant structs and more end-user overload. There would be a lot of conversions though. We would have had 4 auxiliary data types instead of 1 in CSL if we never edited the code, for example.

If we want to not modify it at all I think the best interface we could make would be that there's one single interface on `CIP25Metadata` that is like `get(policyid, assetname) -> Option<details>` (and 1 setter for creating) and then make every other struct in the crate private to hide all those redundant types that should only matter for encoding. Thoughts? Providing a better interface for exploring the full structure is awkward unless we unify the v1 vs v2 policyids/assetnames which unless there's general editing of serialization code to ensure that they're encoded correctly, or would allow you to  end up with a format that's neither v1 nor v2.

Would users want to, say, find all NFTs with a single policy ID? Or do they care more about policyID x assetname pairs? If it's the latter then the simplified wrapper interface is more reasonable (but would still be going between the chain policyid/assetnames and the two v1/v2 ones local to here if we're not editing serialization code).